### PR TITLE
refactor(Core/BG/AV): code cleanup

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
##### CHANGES PROPOSED:

- Fixed some Codacity warnings
- Fixed some clang-tidy warnings
- Other code cleanup

##### TESTS PERFORMED:

None

##### HOW TO TEST THE CHANGES:

- Make sure Battlground Alterac Vally is not broken and there is not regression

##### Target branch(es):

Master

### CODE REVIEW

**Comment by wetbrownsauce** [comment]
Tested a bunch of stuff in AV with this PR. I'm not sure what existed prior to this PR, but here is a big list of what does and doesn't work. If none of the broken stuff was a result of this PR, then just ignore it.


### What Works:

- Announcements all work fine. This includes things like chat announcements for when the BG is about to begin (2min, 1min, 30sec, start), Graveyard assaults and captures, Tower assaults, defends, and destruction, etc. 
- Players cannot leave starting tunnel until battle begins
- You cannot assault something that your faction already controls
- All mobs appear to be spawned
- Statistics in the leaderboard are tracked (tower/gy assaults, etc)
- Reinforcements decrease when a player is killed by another player
- Galvangar and Balinda emote and yell correctly when aggro'd. They appear to do all of their attacks properly, and reset when pulled out of their buildings.
- Destroying towers/bunkers correctly decreases resources of opposing faction
- Areatriggers all seem to work fine
- Player is correctly dismounted when entering indoor areas
- Mines announce when taken by a faction
- Resources slowly increase when controlling a mine
- Resources increase twice as much when controlling both mines
- Respawning through nearest GY works fine
- Horde/Ally bosses lose buffs/guards as matching tower/bunker is destroyed
- Horde/Ally bosses reset if pulled out of keeps
- Horde/Ally bosses emote and yell correctly
- BG ends when Alliance kill the Horde boss.


### What Doesn't Work:

- Starting gate gameobject is sometimes not there before battle begins. Leaving the start tunnel before the battle begins still ports the player back in.
- Irondeep mine does not announce that it is Trogg controlled at start of battle (I am not sure if this is blizzlike, the Coldtooth mine announces at start that Snivvle controls it, but nothing for Irondeep)
- No NPC's have paths or wander
- Stormpike Quartermaster's gossip options do nothing
- Galvangar is facing the wrong direction on spawn
- Horde Wing Commander does not aggro and can't be attacked

- **Server crashes when the Horde defeat the Alliance boss. (See attached image)**
crash


### Untested:

- Does bg end when reinforcements are depleted?
- Does bg announce when reinforcements are low?

**Comment by wetbrownsauce** [comment]
Re-tested WITH the PR this time. It seems that most of the bugs that exist in this PR exist on master and are not a result of this PR. The only difference is that when the PR is applied, worldserver crashes when either faction kills an end boss to end the BG (on master this only happens for the horde killing alliance boss). 

Issue | PR | Master
:---: | :---: | :---:
Start timer works | Yes | Yes
Gate gameobject exists during countdown | Yes? | Sometimes
Tower/Bunker assault/defend/destroy announced | Yes | Yes
GY assault/defend/capture announced | Yes | Yes
Can only assault points controlled by opposing faction | Yes | Yes
All mobs spawned | Yes | Yes
Stats in scoreboard track correctly | Yes | Yes
Reinforcements decrease when a player is killed by another player | Yes | Yes
NPC's killing a player correctly ignore reinforcement decrease | Yes | Yes
Galvangar and Balinda emote and yell correctly | Yes | Yes
Galvangar and Balinda attack and rotate their abilities correctly | Yes | Yes
Galvangar and Balinda reset if pulled out of their huts | Yes | Yes
Destroying towers/bunkers correctly decreases reinforcements | Yes | Yes
Areatriggers work | Yes | Yes
Player is dismounted when indoors | Yes | Yes
Mines announce when taken by a faction | Yes | Yes
Mines grant slow resource gain when controlled | Yes | Yes
Resources gained from mines are doubled if a faction controls both | Yes | Yes
Respawning at the nearest owned GY works | Yes | Yes
Faction bosses lose buffs/guards as matching tower/bunkers are destroyed | Yes | Yes
Faction bosses reset if pulled out of keeps | Yes | Yes
Faction bosses emote/yell correctly | Yes | Yes
BG ends when a faction boss is slain | Crash (both sides) | Crash (only when horde kills ally)
Mines announce who controls them at very start of battle | only coldtooth | only coldtooth
NPC's walk on waypoint paths | No | No
Trash mobs wander around their spawn points | No | No
Stormpike Quartermaster's gossip options work | No | No
Galvangar is facing the right direction when spawned | No | No
Horde wing commander can be fought and killed | No | No

**Comment by Seebaby** [comment]
So I tested the Alterac Vally with 2 players chars Horde and Allianz once killed the Horde bosses and conquered all points. I did the same with the Alliance.

I didn't crash whether I killed the Alliance boss or the Horde boss.

The Bg's always ran about 25-30 min.

**Comment by FrancescoBorzi** [comment]
I don't really see how this change can introduce a crash, I'm almost sure that the crash got by @wetbrownsauce was determined by something else. @Selyria is also confirming that no crash has been introduced here, for which I thank him a lot.

But, just to be safe, it would be nice if someone else can double check this.

**Comment by FrancescoBorzi** [comment]
@wetbrownsauce I've tried to reproduce the crash myself by killing the horde boss using an ally character. I wasn't able to reproduce it. I've used `.debug bg` to do it, no other players involved.

Have you tested with players? If yes, was any player queued for some other BG during that time?

**Comment by wetbrownsauce** [comment]
That was a while ago, but iirc it only occurred when using `.debug bg` and having at least one player on either side.

**Comment by FrancescoBorzi** [comment]
Can anyone test this without `.debug bg`, having the players that are in AV also queued for some other BG ?

**Comment by FrancescoBorzi** [comment]
I've updated this branch so now it includes this fix: https://github.com/azerothcore/azerothcore-wotlk/pull/1931

Probably the crash has been fixed here too. Please test this.

File: src/server/game/Battlegrounds/Zones/BattlegroundAV.h [hunk 1]
```diff
@@ -1633,18 +1632,18 @@ class BattlegroundAV : public Battleground
         Creature* AddAVCreature(uint16 cinfoid, uint16 type);
 
         /*variables */
-        int32 m_Team_Scores[2];
-        uint32 m_Team_QuestStatus[2][9]; //[x][y] x=team y=questcounter
+        int32 m_Team_Scores[2]{};
```
[the remark this answers is not in the document]
Comment (FrancescoBorzi) [reply src/server/game/Battlegrounds/Zones/BattlegroundAV.h]: clang tidy autocorrection. The other line is re-added below.

**Comment by FrancescoBorzi** [comment]
Can anyone test this please? @pklloveyou  ?

> Make sure Battlground Alterac Vally is not broken and there is no regression

**Comment by Endeffect-AT** [comment]
PR needs to be updated to be tested:

Loading Trinity strings...
SQL: SELECT entry, content_default, content_loc1, content_loc2, content_loc3, content_loc4, content_loc5, content_loc6, content_loc7, content_loc8 FROM trinity_string
ERROR: [1146] Table 'acore_world.trinity_string' doesn't exist
Your database structure is not up to date. Please make sure you've executed all queries in the sql/updates folders.

**Comment by FrancescoBorzi** [comment]
@Endeffect-AT thanks a lot, I will update this.

**Comment by FrancescoBorzi** [comment]
@Endeffect-AT done, can you please test this?

**Comment by Endeffect-AT** [comment]
Sure, I'm already there 👍

**Comment by Endeffect-AT** [comment]
Ran Build succesfully on windows 10 pro 64 bit woíthout errors on commit 6fb638c62132.
There was one crash during the alliance test, not sure about the cause but i ithink it was not related to the bg!

Tested BG with command .debug bg on Horde/Alliance side:

---Horde---
----------------------

- [x] Coldtooth Mine captured
- [x] Irondeep Mine captured
- [x] Snowfall Graveyard captured
- [x] Stonehearth Bunker destroyed
- [x] Stonehearth Graveyard captured
- [x] Stormpike Graveyard destroyed 
- [x] Icewing Bunker destroyed
- [x] Dun Baldar South Bunker destroyed
- [x] Dun Baldar North Bunker destroyed
- [x] Storm Pike Aid Station captured
- [x] Killed Captain Balinda Stonehearth
- [x] Killed Alliance Boss (No Crash)


---Alliance---
----------------------

- [x] Irondeep Mine captured
- [x] Snowfall Graveyard captured
- [x] Iceblood Tower destroyed
- [x] Iceblood Graveyard captured
- [x] Tower Point destroyed
- [x] Coldtooth Mine captured
- [x] Frostwolf Graveyard captured
- [x] East Frostwolf Tower destroyed
- [x] West Frostwolf Tower destroyed
- [x] Frostwolf Relief Hut captured
- [x] Killed Captain Galvangar
- [x] Killed Horde Boss (No Crash)

Mechanics & NPC's
-------------------
- [x] Start timer works (2 min, 1 min, 30 sec)
- [x] Gate gameobject exists during countdown
- [x] BG closes after 5 minutes if there are too less players (Tested 1 Horde & 1 Alliance Player)
- [x] Can only assault points controlled by opposing faction
- [x] Graveyard assault/defend/capture announced
- [x] Tower/Bunker assault/defend/destroy announced	
- [x] Player is dismounted when indoors
- [x] Respawning at the nearest owned graveyard works
- [x] Stats in scoreboard track correctly
- [x] Mines announce when taken by a faction
- [x] Mines announce at very start of battle for Coldtooth Mine
- [x] After win closing timer works and player got kicked (2 min, 1 min, 45sec, 30 sec, 15 sec)
- [x] Galvangar and Balinda reset if pulled out of their huts
- [x] Faction bosses lose buffs/guards as matching tower/bunkers are destroyed
- [x] Faction bosses reset if pulled out of keeps
- [x] BG ends when a faction boss is slain
- [x] All mobs spawned (Spawn locations in mines look weird)
- [x] Trash mobs except mobs in mines wander around their spawn points 
- [ ] Trash mobs in mines wander around their spawn points 
- [ ] Mines announce at very start of battle for Irondeep Mine
- [ ] NPC's walk on waypoint paths
- [ ] Stormpike Quartermaster's gossip options work

Not tested because not sure about the correctness:
------------ 
- [ ] All mobs spawned
- [ ] Reinforcements decrease when a player is killed by another player
- [ ] NPC's killing a player correctly ignore reinforcement decrease
- [ ] Galvangar and Balinda emote and yell correctly
- [ ] Galvangar and Balinda attack and rotate their abilities correctly
- [ ] Destroying towers/bunkers correctly decreases reinforcements	
- [ ] Areatriggers work
- [ ] Mines grandt slow resource gain when controlled	
- [ ] Resources gained from mines are dmmoubled if a faction controls both	
- [ ] Faction bosses emote/yell correctly	
- [ ] Galvangar is facing the right direction when spawned
- [ ] Horde wing commander can be fought and killed

**Comment by FrancescoBorzi** [comment]
@Endeffect-AT  thanks a lot!

**Comment by Endeffect-AT** [comment]
**Addition:** For those who read my post yesterday, I accidentally posted the wrong commit hash on which i tested this ... Sry people, sometimes my brain is AFK - lol. 
Of course I tested the PR like every PR with the last commit.

**Pull request merged**